### DISCUSSION TITLE
Bulk create content item?

### DISCUSSION THREAD

**Discussion by orchardc** [OrchardCMS/OrchardCore#7982] [Q&A]
Is it possible to bulk create content items?

I tried the following but it doesn't work.

await _contentManager.ImportAsync(myProducts);

**Comment by Skrypt** [comment]
Technically the YesSQL session already batches request by itself. You can loop on the ContentManager and it is a `no operation call`. It's only executed on the `dispose` event later on. So, nothing special here. Just do a simple do while or for each loop.

If you run into issues let us know. There is also other options.

**Comment by orchardc** [comment]
Thanks @Skrypt , do you mean something like below, using CreateAsync?

```C#
foreach(var contentItem in contentItems)
{
...
   await _contentManager.CreateAsync(contentItem);
}
```
